run-build-functions.sh

```diff
@@ -72,6 +72,29 @@ mkdir -p $NETLIFY_CACHE_DIR/.cargo
 : ${NPM_FLAGS=""}
 : ${BUNDLER_FLAGS=""}
 
+# Run `nvm install ...`
+nvm_install() {
+  if nvm install "$@"; then
+    return 0
+  fi
+
+  # nvm install might fail due to either:
+  #  - A user error, e.g. the NODE_VERSION being an invalid version
+  #  - A third-party outage of nodejs.org/dist
+  # We distinguish between those by making a HEAD HTTP request to nodejs.org/dist
+  if curl --head --fail "$DEFAULT_NODE_MIRROR"; then
+    return 1
+  fi
+
+  NVM_NODEJS_ORG_MIRROR="$FALLBACK_NODE_MIRROR" nvm install "$@"
+}
+
+# Environment variable used by nvm to customize where the Node.js binaries are
+# downloaded from. Our default value matches the one from nvm.
+DEFAULT_NODE_MIRROR="${NVM_NODEJS_ORG_MIRROR:-"https://nodejs.org/dist"}"
+# When nodejs.org/dist/ is down, we try to use a mirror site as a fallback.
+FALLBACK_NODE_MIRROR="https://npmmirror.com/mirrors/node"
```

Comment:
This mirror is actually mostly meant to download Node.js from China.
This is the only mirror for `nodejs.org/dist` I could find. It is much slower, but it works. If you know of any others, please add it so we have multiple fallbacks.